x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -218,7 +218,7 @@ export const calculateEndpointAuthz = (
     canReadSecuritySolution,
     canAccessFleet: fleetAuthz?.fleet.all ?? userRoles.includes('superuser'),
     canAccessEndpointManagement: hasEndpointManagementAccess,
-    canCreateArtifactsByPolicy: hasEndpointManagementAccess && isPlatinumPlusLicense,
+    canCreateArtifactsByPolicy: isPlatinumPlusLicense,
```

Comment:
Not sure about what you mean on this. It is setting the `canCreateArtifactsByPolicy` privilege to `true` when the license is platinum and `false` when not. The code looks exactly as before, just removed the `hasEndpointManagementAccess` check

Reply:
I mean, that if we want to use only the license info for this check then this is not the best place to add it IMO. You should just remove the `canCreateArtifactsByPolicy ` key from `authz`, and use the license service where `canCreateArtifactsByPolicy` is explicitly used now.

so if you have
```
const canCreateArtifactsByPolicy = useUserPrivileges().endpointPrivileges
``` 
it should be replaced with:
```
const licenseService = useLicense();
const hasPlatinumLicense = licenseService.isPlatinumPlus();
```
Consequently, you'd also need to change the [BaseValidator](https://github.com/elastic/kibana/blob/4a3af9c97bb0a9bfaccfae77966625193337f41f/x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/base_validator.ts#L54)
to accept a `licenseService` param that allows you to compute `isAllowedToCreateArtifactsByPolicy` as there would be no `canCreateArtifactsByPolicy` in `endpointAuthzPromise` property.

Reply:
Thanks for the explanation, I understand now what you meant. 

I can see the `canCreateArtifactsByPolicy` as a privilege check, now is based on licence only but it wasn't before and we don't know in the future. 
As it is already in the `authz` service, where we try to keep all the privileges checks, I would let this here and not do this refactor. I like to keep this kind of checks in the same place as it helps us to include/remove check conditions in one place and not in multiple places. 

I'm ok with doing that change but would like to know other team members thoughts 🙂

Reply:
Yeah, this one is a strange one because it is controlling only a small area of functionality on a page. I do agree with Ash that perhaps `EndpointAuthz` is not necessary the best place for it, but it is super convenient to continue to have all authz type of checks in one place (which is also used on the server side). I would vote for keeping it in Authz.

Re: implementation

You could just check `isPlatinumPlus` or you could also code it sot hat it checks that the user has at least one of the artifacts privileges, although not technically needed on this one:

```typescript
canCreateArtifactsByPolicy: isPlatinumPlusLicense && 
  (
    canWriteTrustedApplications || 
    canWriteHostIsolationExceptions || 
    canWriteBlocklist || 
    canWriteEventFilters
  ),
```

Reply:
I like this check better. But yeah not needed in this PR. We can add it later.